### PR TITLE
Add initial package setup

### PR DESCRIPTION
**Pull request by benmvp**
This sets up:
- Initial dependencies
- Babel
- ESLint
- Flow & Flow coverage
- Prettier ESLint on commit
- Jest

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -48,3 +48,6 @@ typings/
 
 # dotenv environment variables file
 .env
+
+# Editor configurations
+.vscode
```
Comment (kaylieEB) [inline .gitignore]: We probably want to add *.log at some point

File: package.json [hunk 2]
```diff
@@ -0,0 +1,82 @@
+{
+    "name": "brite-rest",
+    "version": "0.0.0-semantically-released",
+    "description": "The official JavaScript SDK for the Eventbrite v3 API",
+    "author": "Eventbrite <npmjs@eventbrite.com>",
+    "main": "lib/umd/index.js",
```
Comment (kaylieEB) [inline package.json]: If they're using `main` it means they're using commonjs system, so it shoud just be lib/index.js, right?

File: package.json [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        "eslint-plugin-import": "^2.0.0",
+        "flow-bin": "^0.65.0",
+        "flow-coverage-report": "^0.4.1",
+        "husky": "^0.14.3",
+        "jest": "^22.3.0",
+        "lint-staged": "^6.1.0",
+        "node": "^8.9.2",
+        "npm-run-all": "^4.1.2",
+        "prettier-eslint-cli": "^4.7.1"
+    }
```
Comment (kaylieEB) [inline package.json]: Could be a separate pr, but what do you think about adding `bundleSize`? https://github.com/siddharthkp/bundlesize
Seems to be a popular option with third party libraries and would keep our bundle sizes in check

Comment (benmvp) [reply package.json]: Oooooh! Nice! Yeah I'll add that in a separate PR when we actually are building a dist

File: .gitignore (hunk 1 above)
Comment (benmvp) [reply .gitignore]: Good point. I'll add when it comes up

File: package.json (hunk 2 above)
Comment (benmvp) [reply package.json]: We have `lib/umd` (line 6) and `lib/esm` (lines 7 & 8). So I think we need the subfolders
